packages/api/src/external/hie/tally-doc-query-progress-with-source.ts

```diff
@@ -0,0 +1,131 @@
+import { DocumentQueryProgress, Progress } from "@metriport/core/domain/document-query";
+import { getStatusFromProgress } from "@metriport/core/domain/document-query";
+import { MedicalDataSource } from "@metriport/core/external/index";
+import { PatientExternalData } from "@metriport/core/domain//patient";
+import { ProgressType } from "@metriport/core/domain/document-query";
+import { Patient } from "@metriport/core/domain/patient";
+import { PatientModel } from "../../models/medical/patient";
+import { executeOnDBTx } from "../../models/transaction-wrapper";
+import { getPatientOrFail } from "../../command/medical/patient/get-patient";
+import { handleWebhookBeingSent } from "../../command/medical/document/process-doc-query-webhook";
+import { PatientDataCommonwell } from "../commonwell/patient-shared";
+import { PatientDataCarequality } from "../carequality/patient-shared";
+import { aggregateAndSetHieProgresses } from "./set-doc-query-progress-with-source";
+
+export type HIEPatientData = PatientDataCommonwell | PatientDataCarequality;
+
+type DynamicProgress = Pick<Progress, "successful" | "errors">;
+
+export type SetDocQueryProgressWithSource = {
+  source: MedicalDataSource;
+  downloadProgress?: DynamicProgress | undefined | null;
+  convertProgress?: DynamicProgress | undefined | null;
+  patient: Pick<Patient, "id" | "cxId">;
+  requestId: string;
+};
+
+/**
+ * Updates the successful and error count for the given hie which is then aggregated
+ * to the patient's document query progress.
+ * @returns
```

Comment:
Can we add a note referring to `setDocQueryProgressWithSource` to update `successful`/`errors`?

nit: empty line between description and params/returns as part of TS/JSDoc

Reply:
You mean `setDocQueryProgressWithSource ` to update `total/status`?